src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -821,15 +885,19 @@ async def handle_member_joined_channel(ack, user, body, client, db_session, cont
 
     incident = incident_service.get(db_session=db_session, incident_id=context["subject"].id)
 
+    # TODO: (wshel) check this
     if body["inviter"]:
         inviter_email = await get_user_email_async(client=client, user_id=body["inviter"])
-        client.user
         added_by_participant = participant_service.get_by_incident_id_and_email(
             db_session=db_session, incident_id=context["subject"].id, email=inviter_email
         )
         participant.added_by = added_by_participant
         participant.added_reason = body["text"]
     else:
+        inviter_email = await get_user_email_async(client=client, user_id=body["inviter"])
+        added_by_participant = participant_service.get_by_incident_id_and_email(
+            db_session=db_session, incident_id=context["subject"].id, email=inviter_email
+        )
         participant.added_by = incident.commander
```

Comment:
Should this be: `participant.added_by = added_by_participant`

Reply:
This function looks a bit different in general than the original one: https://github.com/Netflix/dispatch/blob/a9045d63419b34de4402ffb8a80e1833de32669f/src/dispatch/plugins/dispatch_slack/events.py#L329

Reply:
There are two general cases we have to cover:

1) Someone @'s a user into the incident channel - In this case, we have an `inviter` and we can cleanly assign the `added_by` field
2) Someone adds themselves via the "join" button. 

Looking at this function, we should test the "join" button as it seems the case were `inviter` is unavailable isn't correctly covered.

Reply:
Yeah this is broken, i'll fix the latter case.

Reply:
While trying to resolve the `handle_member_joined_channel` function, I think I found a separate issue w/ `@app.event` specific functions that I am stumped on. 

The `@app.event` functions (`handle_member_join/left`) never trigger (based on simply adding a debug print inside these functions) and a `BoltUnhandledRequestError` makes its way through the global error handler. 

I've tried:

- removing all the middleware specified in the `app.event` decorator
- changing `@app.event("member_joined", ...` -> `@app.event({"type": "member_left"}, ...`
- adding an `ack()` to all functions that looked to be missing one (`handle_timeline_added_event`, `handle_thread_creation`, `handle_message_tagging`, etc)
- checking for BoltUnhandledRequestError as the type for error in the global error handler and returning a 200 if it is, re: https://github.com/slackapi/bolt-python/issues/299
- adding a [trivial](https://github.com/slackapi/bolt-python/blob/main/examples/message_events.py#L19) @app.event function that just replies back, which didn't work (code isn't executed)

No exception is raised in the logs. But, here's what they look like around the `member_left_channel` event happening:

```bash
DEBUG:Checking listener: handle_workflow_list_command ...:/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_bolt/app/async_app.py:async_dispatch:526
DEBUG:Checking listener: handle_workflow_run_command ...:/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_bolt/app/async_app.py:async_dispatch:526
INFO:Request body: {'token': 'XYZ', 'team_id': 'T04FZTFLBFE', 'api_app_id': 'A04FGTKNP2B', 'event': {'type': 'member_left_channel', 'user': 'U04FXDYTDMY', 'channel': 'C04FZTBDGGP', 'channel_type': 'C', 'team': 'T04FZTFLBFE', 'event_ts': '1673309703.001400'}, 'type': 'event_callback', 'event_id': 'Ev04J4TCMKBM', 'event_time': 1673309703, 'authorizations': [{'enterprise_id': None, 'team_id': 'T04FZTFLBFE', 'user_id': 'U04FXC3VC75', 'is_bot': True, 'is_enterprise_install': False}], 'is_ext_shared_channel': False, 'event_context': '4-XYZ'}:/Users/wshel/Projects/dispatch/src/dispatch/plugins/dispatch_slack/bolt.py:app_error_handler:41
ERROR:Error: :/Users/wshel/Projects/dispatch/src/dispatch/plugins/dispatch_slack/bolt.py:app_error_handler:44
NoneType: None
LOG:{'error': BoltUnhandledRequestError(), 'client': <slack_sdk.web.async_client.AsyncWebClient object at 0x153ebf520>, 'body': {'token': 'XYZ', 'team_id': 'T04FZTFLBFE', 'api_app_id': 'A04FGTKNP2B', 'event': {'type': 'member_left_channel', 'user': 'U04FXDYTDMY', 'channel': 'C04FZTBDGGP', 'channel_type': 'C', 'team': 'T04FZTFLBFE', 'event_ts': '1673309703.001400'}, 'type': 'event_callback', 'event_id': 'Ev04J4TCMKBM', 'event_time': 1673309703, 'authorizations': [{'enterprise_id': None, 'team_id': 'T04FZTFLBFE', 'user_id': 'U04FXC3VC75', 'is_bot': True, 'is_enterprise_install': False}], 'is_ext_shared_channel': False, 'event_context': '4-XYZ'}, 'logger': <Logger slack_bolt.AsyncApp (DEBUG)>}
INFO:Unsuccessful Bolt execution result (status: 500, body: {"token": "XYZ", "team_id": "T04FZTFLBFE", "api_app_id": "A04FGTKNP2B", "event": {"type": "member_left_channel", "user": "U04FXDYTDMY", "channel": "C04FZTBDGGP", "channel_type": "C", "team": "T04FZTFLBFE", "event_ts": "1673309703.001400"}, "type": "event_callback", "event_id": "Ev04J4TCMKBM", "event_time": 1673309703, "authorizations": [{"enterprise_id": null, "team_id": "T04FZTFLBFE", "user_id": "U04FXC3VC75", "is_bot": true, "is_enterprise_install": false}], "is_ext_shared_channel": false, "event_context": "4-XYZ"}):/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_bolt/adapter/socket_mode/async_internals.py:send_async_response:45
DEBUG:Message processing completed (type: events_api, envelope_id: 7632b277-354a-48ba-b5a1-f217a546e6db, session: s_326826156):/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_sdk/socket_mode/async_client.py:run_message_listeners:163
DEBUG:Received message (type: PONG, data: bytearray(b'sdk-ping-pong:1673310079.643423'), extra: , session: s_326826156):/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_sdk/socket_mode/aiohttp/__init__.py:receive_messages:243
DEBUG:Received message (type: PING, data: bytearray(b'Ping from applink-4'), extra: , session: s_326826156):/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_sdk/socket_mode/aiohttp/__init__.py:receive_messages:243
```

Here's a current version of my global error handler function:

```python
@app.error
async def app_error_handler(
    error: Any, client: AsyncWebClient, body: dict, logger: logging.Logger
) -> BoltResponse:

    if body:
        logger.info(f"Request body: {body}")

    if error:
        logger.exception(f"Error: {error}")

    if body.get("view"):
        modal = Modal(
            title="Error", close="Close", blocks=[Section(text="Something went wrong...")]
        ).build()

        await client.views_update(
            view_id=body["view"]["id"],
            view=modal,
        )

    print("LOG:" + str(locals()))
    return BoltResponse(body=body, status=HTTPStatus.INTERNAL_SERVER_ERROR.value)
```

I probably have tunnel vision on this by now and am missing something obvious. Any thoughts on what could be preventing the `handle_member_joined/left_channel` functions (or [really any](https://github.com/slackapi/bolt-python/blob/main/examples/message_events.py#L19) `app.event` function) from executing?

Reply:
I would disable the entire `@app.error` function. I think there is an open issue where the bolt swallows actual exceptions.

Reply:
Ah yep, that did it. It was just the wrong event name in this case. But yeah the error handler is eating the errors. tyty.

Reply:
This should be resolved now, let me know if you spot any issues in the new implementation.